sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1976,6 +1976,17 @@ val SHUFFLE_COMPRESSION_LZ4_CHUNK_SIZE = conf("spark.rapids.shuffle.compression.
       .integerConf
       .createWithDefault(20)
 
+  val SHUFFLE_PARTITIONING_MAX_CPU_BATCH_SIZE =
+    conf("spark.rapids.shuffle.partitioning.maxCpuBatchSize")
+      .doc("The maximum size of a sliced batch output to the CPU side " +
+        "when GPU partitioning shuffle data. This is used to limit the peak on-heap memory used " +
```

Comment:
So one thing I would say was that this wasn't introduced to limit peak memory per se, it was introduced because we cannot go above this number (https://github.com/NVIDIA/spark-rapids/issues/45). If we are going to make it configurable, we should perhaps add a max check so we don't go above it. The comment could reflect this constraint.

+1 on the internal comment from @winningsix

Reply:
fixed